dotCMS/src/integration-test/java/com/dotmarketing/util/ConfigTest.java

```diff
@@ -0,0 +1,254 @@
+            
+        }
+ 
+        
+    }
+    
+
+    @Test
+    public void testing_notfound_booean_returns() {
+
+        try {
+            Config.getBooleanProperty("no-property");
+            assert(false);
+        }
+        catch(Exception e) {
+            assert(e instanceof NoSuchElementException);
+            
+        }
+ 
+        
+    }
+    
+    
+    
+    @Test
+    public void test_get_integer_from_env() {
+
+
+        int value =Config.getIntProperty("no-property", -99);
+        assert(value==-99);
+        
+        // this should not work, as we prefix DOT_ to the env variable lookup
+        value =Config.getIntProperty(XXX_TESTING_INTEGER, -99);
+        assert(value==-99);
+        
+        // we should get back Integer.MAX_VALUE, as this get transformed into DOT_TESTING_INTEGER
+        value =Config.getIntProperty("testing.integer", -99);
+        assert(value==Integer.MAX_VALUE);
+    }
+
+    @Test
+    public void test_get_float_from_env() {
+
+
+        float value =Config.getFloatProperty("no-property", 3.14f);
+        assert(value==3.14f);
+        
+        // this should not work, as we prefix DOT_ to the env variable lookup
+        value =Config.getFloatProperty(XXX_TESTING_INTEGER, 3.14f);
+        assert(value==3.14f);
+        
+        // we should get back Integer.MAX_VALUE, not the default
+        value =Config.getFloatProperty("testing_float",-1f);
+        assert(value==Float.MAX_VALUE);
+    }
+
+    
+    @Test
+    public void test_get_string_from_env() {
+
+
+ 
+        String value =Config.getStringProperty("no-property", TESTING_VALUE);
+        assert(value.equals(TESTING_VALUE));
+        
+        // this should  work, as we prefix DOT_ to the env variable lookup
+        value =Config.getStringProperty("testing.integer", TESTING_VALUE);
+        assert(value.equals(String.valueOf(Integer.MAX_VALUE)));
+        
+        // we should get back VALUE_ABC
+        value =Config.getStringProperty("testing.string","VALUE_ABC");
+
+    }
+    
+    @Test
+    public void test_get_string_array_from_env() {
+
+
+ 
+        String[] value =Config.getStringArrayProperty("no-property", TESTING_ARRAY);
+        assertArrayEquals(value, TESTING_ARRAY);
+        
+        // this should  work, as we prefix DOT_ to the env variable lookup
+        value =Config.getStringArrayProperty("testing_string_with_comma");
+        assert(value.length==2);
+        assert(value[1].equals("VALUE2"));
+        
+        String notArray  =Config.getStringProperty("testing_string_with_comma");
+        assertEquals(notArray, "VALUE1,VALUE2");
+    }
+    
+    
+    /*
+     * 
+     * Restore default variables for each test
+     */
+    @AfterClass
+    public static void resetMap() {
+        envMap.clear();
+        envMap.putAll(DEFAULTS);
+    }
+    
+    
+
+    
+    
+    
+    @BeforeClass
+    public static void accessFields() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5943250)